distributed/scheduler.py

```diff
@@ -1590,6 +1596,33 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    def publish_dataset(self, stream=None, keys=None, data=None, name=None,
+                        client=None):
+        self.client_wants_keys(keys, 'published-%s' % name)
+        self.published_data[name] = {'data': data, 'keys': keys,
+                                     'clients': {client}}
```

Comment:
Then the best would be passing back a result "succeeded" - in fact, that flag could be passed back as-is to the user without an exception.